### ISSUE TITLE
Allow users to license their posts and comments

### ISSUE THREAD

**Issue by h3ndrik** [LemmyNet/lemmy#4298]
### Requirements

- [X] Is this a feature request? For questions or discussions use https://lemmy.ml/c/lemmy_support
- [X] Did you check to see if this issue already exists?
- [X] Is this only a feature request? Do not put multiple feature requests in one issue.
- [X] Is this a backend issue? Use the [lemmy-ui](https://github.com/LemmyNet/lemmy-ui) repo for UI / frontend issues.

### Is your proposal related to a problem?

Allow users to license their content, similar to choosing the language for each post.

Currently user posts and comments don't have licensing information attached to them. For federation and other instances it would be beneficial to know how content is licensed. We discussed that in the context of Meta's Threads "stealing" our content (or using it consentually, that isn't clear without licensing). And I regularly see people ask who "owns" their posts. Or if it can be used to train AI. It would also be necessary if wikis or other forms of content get implemented.

Free- / Libre- culture will also benefit from that. Sharing and collaborating should be encouraged and we need licensing to do that with the content here. As Lemmy is free software and part of that ecosystem, I'd like it to participate.

### Describe the solution you'd like.

I'd like to offer users the ability to license their content. Next to the drop-down where you can choose the language of the text. Additionally a user should be able to select a default license for their content in their user settings. Some sane choices should be offered, maybe also a text input. That would be the user facing AI. The license needs to be stored and made available in ActivityPub alongside the actual content. (Don't make it an overwhelming amount of options to choose from, the creative commons licenses, public domain and maybe the gnu free documentation license suffice, plus an option that content isn't freely licensed.)

### Describe alternatives you've considered.

All user-content remains copyrighted.

### Additional context

There has been an old and closed feature request here: https://github.com/LemmyNet/lemmy/issues/172
I'd like to re-request it since circumstances have changed.

**Comment by Nutomic** [comment]
Having a different license for each user would cause too much complexity and fragmentation. It makes more sense to have a license per instance. This can already be configured and gets displayed under /legal. Then based on the legal info users can decide where to signup, and admins can decide which instances to federate with.

**Comment by h3ndrik** [comment]
Feel free to close this proposal if deemed too complex. I don't think the /legal page can be taken as binding, as users don't explicitly accept it or read it in the first place. Additionally users may post different content with different licensing or copy content. So I wouldn't rely on anything that is set per instance in the fine-print. Additionally there is the federation aspect. We also display content that isn't from the same instance and those people don't abide by the same rules.

**Comment by Nutomic** [comment]
The legal text could be shown on the signup page. And I dont really see why youd want to use a different license for each post. Maybe when posting on separate communities, but then you could also create separate accounts on different instances for each license.

Also its not clear to me that the license would really have much effect. Sure you can mark your comment as copyrighted, but Lemmy and other platforms will still happily federate it around. Anyway the content is treated as public and free to share, regardless of any explicit license.

**Comment by h3ndrik** [comment]
My use-case is a bit the other way around. I like free information, copyleft, share remix and use creative works. With that I need explicit permission to do so. And it's not enough to have some vague idea. Also like with the semantic web it's nice to have meta-information in a machine-readable format attached to that information and not somewhere in the fine-print that needs to be processed by a human and separately for each individual case.

And I myself sometimes post my own creative work, pictures, short stories, Linux how-tos and guides to obscure hardware and I like to share them for others to use freely. All while I also sometimes re-post memes I found somewhere, summarize news articles, scientific papers or copy-paste paragraphs. I don't have permission to re-license that, it might just be subject to fair use. Similarly, I see people writing things themselves, but sometimes they copy whole news articles because the source is pay-walled. This goes against copyright and communities have started telling people not to do it.

With all of that diversity, I can just treat every post as copyrighted or possibly "stolen", and never reuse anything from Lemmy.

**Comment by Nutomic** [comment]
I see where youre coming from, but to be honest I doubt that most users would bother to set an explicit license for each post (with language tags its also not working that well). Sure there could be a default license defined in some way, but most likely people will regularly forget to set it when they copy-paste a news article and then youre back at square one.

**Comment by h3ndrik** [comment]
Agree. I mean this is just a proposal. And I think it's a good thing if it works that way. The user shouldn't be forced to bother. If they don't... it's alike the current situation. We need a safe fallback for the person sitting at the other end, so the default setting should be '<unspecified>' license. If they forget, that's makes it alright and nothing gets attributed incorrectly. We can just have a benefit in the situation when someone clicks on it anyways.

I included this proposal a bit early since some discussion about Meta happened and it'd be nice to have it already in place when other content types or Wikis get discussed. And it'd be nice for the current content, too. Currently I can live with typing it out underneath my posts. The user-base of Lemmy seems quite diverse. So while Lemmy is free software, it doesn't deliberately make an effort to teach it's users those values. I'd like it to be that way. And I can see it become a necessity in the future due to federation, companies like Meta entering the stage or new content types being introduced. However I can't give any advise on where to put it on the backlog of requested features.

**Comment by dessalines** [comment]
I'll re-open if someone wants to work on this, but there's been no interest.
